qiita_db/support_files/patches/38.sql

```diff
@@ -0,0 +1,241 @@
+
+-- The workflows need to connect the different outputs of a processing job with
+-- the inputs of the next processing job. The following table holds which
+-- artifact was generated in each named output. So we can backtrack and perform
+-- the correct connections when executing the workflow. Note that this information
+-- is only needed for the wokflows, so there is no necessity to populate the
+-- table with all the artifacts that has been already generated. Furthermore,
+-- there is no way to retrieve this information once the job has been executed
+-- and be 100% sure that we are connecting the jobs and the artifacts correctly
+CREATE TABLE qiita.artifact_output_processing_job (
+	artifact_id          bigint  NOT NULL,
+	processing_job_id    uuid    NOT NULL,
+	command_output_id    bigint  NOT NULL
+ ) ;
+CREATE INDEX idx_artifact_output_processing_job_artifact ON qiita.artifact_output_processing_job ( artifact_id ) ;
+CREATE INDEX idx_artifact_output_processing_job_job ON qiita.artifact_output_processing_job ( processing_job_id ) ;
+CREATE INDEX idx_artifact_output_processing_job_cmd ON qiita.artifact_output_processing_job ( command_output_id ) ;
+ALTER TABLE qiita.artifact_output_processing_job ADD CONSTRAINT fk_artifact_output_processing_job_artifact FOREIGN KEY ( artifact_id ) REFERENCES qiita.artifact( artifact_id )    ;
+ALTER TABLE qiita.artifact_output_processing_job ADD CONSTRAINT fk_artifact_output_processing_job_job FOREIGN KEY ( processing_job_id ) REFERENCES qiita.processing_job( processing_job_id )    ;
+ALTER TABLE qiita.artifact_output_processing_job ADD CONSTRAINT fk_artifact_output_processing_job_cmd FOREIGN KEY ( command_output_id ) REFERENCES qiita.command_output( command_output_id )    ;
+
+-- In order to successfully represent the current status of a job,
+-- we need to identify if the job is part of a workflow in construction
+-- and if the job is waiting for a previous job to finish in order to be executed
+INSERT INTO qiita.processing_job_status (processing_job_status, processing_job_status_description)
+	VALUES ('in_construction', 'The job is one of the source nodes of a workflow that is in construction'),
+		   ('waiting', 'The job is waiting for a previous job in the workflow to be completed in order to be executed.');
+
+-- Populate the newly created tables
+DO $do$
+DECLARE
+	in_slq_param_id		bigint;
+	in_sl_param_id		bigint;
+	in_po_param_id		bigint;
+	dflt_slq_id			bigint;
+	dflt_sl_id			bigint;
+	dflt_per_sample_id	bigint;
+	dflt_po_id			bigint;
+BEGIN
+	-- Add the artifact type information for the input parameters for the commands
+	-- command_id = 1 -> Split libraries FASTQ
+	SELECT command_parameter_id FROM qiita.command_parameter
+		WHERE command_id = 1 AND parameter_name = 'input_data'
+		INTO in_slq_param_id;
+
+	-- Split libraries FASTQ supports FASTQ (3) and per_sample_FASTQ (5)
```

Comment:
Yes - those are the ones that we care about here. Note that the artifact types is a controlled vocab. That is, all Qiita databases should have the same values ATM, since it is only changed i the patches.